pandas/tests/io/json/test_pandas.py

```diff
@@ -59,7 +60,8 @@ def setup_method(self, method):
         self.mixed_frame = _mixed_frame.copy()
         self.categorical = _cat_frame.copy()
 
-    def teardown_method(self, method):
+        yield
```

Comment:
I learned that autouse fixtures can include `yield`, letting us do the tearndown here which is nice.

Reply:
On the other hand, it also makes this more "complex" to understand IMO (the `teardown_method` is rather explicit and easy to understand what is going on, while the `yield` is a less standard fancy feature of pytest ). 
Has the `autouse=True` other advantages? (what's the reason you needed that here?)

Reply:
Two things

1. `datapath` can only be called inside a test, since it inspects the `request` context, to see if `--strict-data-files` is set.
2. datapath is a fixture, and you can't mix xUnit style `setup_method` with fixtures. `autouse` fixtures are the easiest way to replicate xUnit stuff, while being able to use fixtures.

Reply:
I've done this myself a few times so wanted to chime in. `autouse=True` allows the fixture to yield every test within it's scope without having to explicitly decorating or passing that fixture as an arg.

This approach is more in line with how pytest suggests doing setup/teardown (see [here](https://docs.pytest.org/en/2.7.3/fixture.html#fixture)) so I think that's a +1 for it. It also gives you potential visibility into the context of the yield tests (ex: here I think you could replace the `datapath` fixture usage as an argument via an assignment to `self.datapath` before yielding)

Reply:
OK, that's a good reason to use an autouse fixture here in this case.

(you don't need to convince me of the benefit of fixtures in general :), however, I think many people are not that familiar with all those pytest special features and it *has* a steeper learning curve IMO, so there can be a balance in how fancy we go)

Reply:
Yeah. When documenting this, I'll recommend against autouse for cases like this. I think it'd be better to just have a `categorical` fixture and accept that in the test. But that would have meant a lot more potential for breaking things :)

Reply:
yeah we should really not use this pattern, rather changing to all fixtures. as a temporary workaround this ok, can you create an issue to 'fix' this properly though.